azurerm/resource_arm_service_fabric_cluster_test.go

```diff
@@ -234,6 +235,41 @@ func TestAccAzureRMServiceFabricCluster_readerAdminClientCertificateThumbprint(t
 	})
 }
 
+func TestAccAzureRMServiceFabricCluster_azureActiveDirectory(t *testing.T) {
+	resourceName := "azurerm_service_fabric_cluster.test"
+	ri := acctest.RandInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMServiceFabricClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMServiceFabricCluster_azureActiveDirectory(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMServiceFabricClusterExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "certificate.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "certificate.0.thumbprint", "33:41:DB:6C:F2:AF:72:C6:11:DF:3B:E3:72:1A:65:3A:F1:D4:3E:CD:50:F5:84:F8:28:79:3D:BE:91:03:C3:EE"),
+					resource.TestCheckResourceAttr(resourceName, "certificate.0.x509_store_name", "My"),
+					resource.TestCheckResourceAttr(resourceName, "azure_active_directory.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "azure_active_directory.tenant_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "azure_active_directory.cluster_application_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "azure_active_directory.client_application_id"),
```

Comment:
after running the tests these fail because we're not checking the first element e.g.
```suggestion
					resource.TestCheckResourceAttrSet(resourceName, "azure_active_directory.0.client_application_id"),
```

will push a commit to resolve this